pinot-core/src/main/java/org/apache/pinot/core/segment/store/SingleFileIndexDirectory.java

```diff
@@ -339,9 +344,92 @@ public void close()
 
   @Override
   public void removeIndex(String columnName, ColumnIndexType indexType) {
-    throw new UnsupportedOperationException(
-        "Index removal is not supported for single file index format. Requested colum: " + columnName + " indexType: "
-            + indexType);
+    if (indexType != ColumnIndexType.INVERTED_INDEX) {
```

Comment:
why limit to INVERTED_INDEX?

Reply:
Yeah, we could refactor/restructure such that it becomes a generic repackaging of indexes (e.g. drop any column/index, add new (??))